src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -104,641 +105,695 @@
+            args.add("delete");
+            args.add("-f");
+            if (deleteConfig) {
+                args.add("-a");
+            }
+            if (kind.equals(ComponentKind.S2I)) {
+                args.add("--s2i");
+            }
+            execute(new File(context), command, envVars, args.toArray(new String[0]));
+        } else {
+            args.add("delete");
+            args.add("-f");
+            args.add("--project");
+            args.add(project);
+            args.add("--app");
+            args.add(application);
+            args.add(component);
+            if (deleteConfig) {
+                args.add("-a");
+            }
+            if (kind.equals(ComponentKind.S2I)) {
+                args.add("--s2i");
+            }
+            execute(command, envVars, args.toArray(new String[0]));
+        }
+    }
+
+    @Override
+    public void undeployComponent(String project, String application, String context, String component, ComponentKind kind) throws IOException {
+        undeployComponent(project, application, context, component, false, kind);
+    }
+
+    @Override
+    public void deleteComponent(String project, String application, String context, String component, ComponentKind kind) throws IOException {
+        undeployComponent(project, application, context, component, true, kind);
+    }
+
+    @Override
+    public void follow(String project, String application, String context, String component) throws IOException {
+        ExecHelper.executeWithTerminal(new File(context), envVars, command, "log", "-f");
+    }
+
+    @Override
+    public void log(String project, String application, String context, String component) throws IOException {
+        ExecHelper.executeWithTerminal(new File(context), envVars, command, "log");
+    }
+
+    @Override
+    public void createProject(String project) throws IOException {
+        execute(command, envVars, "project", "create", project);
+    }
+
+    @Override
+    public void deleteProject(String project) throws IOException {
+        execute(command, envVars, "project", "delete", project, "-f");
+    }
+
+    @Override
+    public void login(String url, String userName, char[] password, String token) throws IOException {
+        if (token == null || token.isEmpty()) {
+            execute(command, envVars, "login", url, "-u", userName, "-p", String.valueOf(password), " --insecure-skip-tls-verify");
+        } else {
+            execute(command, envVars, "login", url, "-t", token, " --insecure-skip-tls-verify");
+        }
+    }
+
+    @Override
+    public void logout() throws IOException {
+        execute(command, envVars, "logout");
+    }
+
+    private static List<Application> parseApplications(String json) throws IOException {
+        JSonParser parser = new JSonParser(JSON_MAPPER.readTree(json));
+        return parser.parseApplications();
+    }
+
+    @Override
+    public List<Application> getApplications(String project) throws IOException {
```

Comment:
@jeffmaury @dgolovin : this temporary workaround is to bypass odo issues for application lists with devfiles

Reply:
@sbouchet :+1:

Reply:
@sbouchet this approach can be used to build application view model when no cluster present. Currently view shows nothing when not connected and logged into the cluster.

Reply:
yes, if odo can get the list of apps without being logged in, this can be a good enhancement.

Reply:
@dgolovin : just discovered that workaround actually does not work as expected, because it only list apps for the current project set in the kubeconfig. 
Meaning that if there is multiple projects in the cluster, only the apps from the current one will be listed by this command... 
so we're still stuck and have to wait for odo to fix issues. 
`      --all-apps         List all components from all applications for the current set project`